eixampledb-core/src/main/java/com/eixampledb/core/impl/EixampleDbMapImplementation.java

```diff
@@ -25,20 +21,59 @@ public GetResponse get(GetRequest getRequest) {
 
     @Override
     public SetResponse set(SetRequest setRequest) {
-        EixampleDbEntry newEntry = map.compute(setRequest.getKey(), (key, entry) -> {
-            long updateTimestamp = System.currentTimeMillis();
-            if (entry != null) {
-                updateTimestamp = entry.getLastupdateTimestamp();
-            }
-           return new EixampleDbEntry(setRequest.getKey(), setRequest.getValue(), updateTimestamp, System.currentTimeMillis(), setRequest.getType());
-        });
+        Object value = setRequest.getType().isNumber() ? NumberUtils.parse(setRequest.getValue()) : setRequest.getValue();
+        EixampleDbEntry newEntry = map.compute(setRequest.getKey(), (key, entry) -> new EixampleDbEntry(
+                setRequest.getKey(),
+                value,
+                creationTimestamp(entry),
+                System.currentTimeMillis(),
+                setRequest.getType()
+        ));
         return new SetResponse(setRequest, true, newEntry);
     }
 
+    @Override
+    public IncrResponse incr(IncrRequest request) {
+        EixampleDbEntry newEntry = map.computeIfPresent(request.getKey(), (key, entry) -> new EixampleDbEntry(
```

Comment:
Now, this operation is atomic. Within the `computeIfPresent` the computation is carried on. `compute` and `computeIfPresent` are thread safe scopes.